### PR TITLE
Including sub-functions (and their actions) as part of the resource explorer

### PR DESCRIPTION
**Pull request by kiiadi**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Related Issue(s)

#410 #402 

## Screenshots (if appropriate)
screen shot 2018-11-07 at 5 38 42 pm

## Checklist

- [x] I have read the **README** document
- [x] I have read the **CONTRIBUTING** document
- [x] Local run of `gradlew check` succeeds
- [x] My code follows the code style of this project
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed
- [ ] A short description of the change has been added to the **CHANGELOG**

## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/Lambda.kt [hunk 1]
```diff
@@ -78,4 +79,26 @@ fun UpdateFunctionConfigurationResponse.toDataClass(credentialProviderId: String
     role = IamRole(this.role()),
     credentialProviderId = credentialProviderId,
     region = region
-)
\ No newline at end of file
+)
+
+fun GetFunctionConfigurationResponse.toConfiguration() = FunctionConfiguration.builder()
```
Comment (abrooksv) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/Lambda.kt]: We end up just coveting this onto a LambdaFunction in the init of the Lambda node, should we refactor LambdaNode to take a LambdaFunction and save us this middle conversion?

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/applications/ServerlessExplorerNodes.kt [hunk 2]
```diff
@@ -25,23 +33,50 @@ class ServerlessApplicationsNode(project: Project) : AwsExplorerPageableNode<Str
...
+            .filter { it.resourceType() == LAMBDA_FUNCTION_TYPE && it.resourceStatus() in COMPLETE_RESOURCE_STATES }
+
     private fun paginationNodeIfRequired(nextToken: String?) = when {
         nextToken != null -> listOf(AwsTruncatedResultNode(this, nextToken))
         else -> emptyList()
     }
+
+    private companion object {
+        val COMPLETE_RESOURCE_STATES = setOf(ResourceStatus.CREATE_COMPLETE, ResourceStatus.UPDATE_COMPLETE)
+        val DELETING_STACK_STATES = setOf(StackStatus.DELETE_IN_PROGRESS, StackStatus.DELETE_COMPLETE)
```
Comment (zhangzhx) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/applications/ServerlessExplorerNodes.kt]: If the deletion failed, the status of DELETE_IN_PROGRESS would be changed to DELETE_FAILED. This gives customers a weird experience that a stack would be shown again after the status change.

Should we show the stacks in DELETE_IN_PROGRESS as well?

Comment (kiiadi) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/applications/ServerlessExplorerNodes.kt]: I think showing `DELETE_FAILED` is fine - the stack is still there - it just might not be in a 'clean' state. Long term I'd love to find a way of representing the status at the top level (red text maybe?).

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/Lambda.kt (hunk 1 above)
Comment (kiiadi) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/Lambda.kt]: 👍

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/applications/ServerlessExplorerNodes.kt (hunk 2 above)
Comment (abrooksv) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/applications/ServerlessExplorerNodes.kt]: I would say grey small text like how the Module says the location when you hover next to the node in grey

Comment (kiiadi) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/applications/ServerlessExplorerNodes.kt]: have opened #432 to track this - other changes have been fixed.

**Pull request merged**